backend/elasticsearch/src/main/java/org/hibernate/search/backend/elasticsearch/schema/management/impl/ElasticsearchIndexSchemaExportImpl.java

```diff
@@ -0,0 +1,99 @@
+/*
+ * Hibernate Search, full-text search for your domain model
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.search.backend.elasticsearch.schema.management.impl;
+
+import java.io.IOException;
+import java.io.OutputStream;
+import java.io.OutputStreamWriter;
+import java.lang.invoke.MethodHandles;
+import java.nio.charset.StandardCharsets;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.function.Consumer;
+
+import org.hibernate.search.backend.elasticsearch.client.spi.ElasticsearchRequest;
+import org.hibernate.search.backend.elasticsearch.logging.impl.Log;
+import org.hibernate.search.backend.elasticsearch.schema.management.ElasticsearchIndexSchemaExport;
+import org.hibernate.search.util.common.logging.impl.LoggerFactory;
+
+import com.google.gson.Gson;
+import com.google.gson.JsonArray;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import com.google.gson.stream.JsonWriter;
+
+public class ElasticsearchIndexSchemaExportImpl implements ElasticsearchIndexSchemaExport {
+	private static final Log log = LoggerFactory.make( Log.class, MethodHandles.lookup() );
+
+	private final Gson userFacingGson;
+	private final String indexName;
+	private final Optional<String> backendName;
+	private final ElasticsearchRequest request;
+
+	public ElasticsearchIndexSchemaExportImpl(Gson userFacingGson, String indexName, Optional<String> backendName,
+			ElasticsearchRequest request) {
+		this.userFacingGson = userFacingGson;
+		this.indexName = indexName;
+		this.backendName = backendName;
+		this.request = request;
+	}
+
+	@Override
+	public Optional<String> backendName() {
+		return backendName;
+	}
+
+	@Override
+	public String indexName() {
+		return indexName;
+	}
+
+	@Override
+	public void write(OutputStream stream) throws IOException {
+		new SimpleExporter().accept( stream );
+	}
+
+	@Override
+	public Map<String, String> parameters() {
+		return Collections.unmodifiableMap( request.parameters() );
+	}
+
+	@Override
+	public List<JsonObject> bodyParts() {
+		return Collections.unmodifiableList( request.bodyParts() );
+	}
+
+	private class SimpleExporter implements Consumer<OutputStream> {
+		@Override
+		public void accept(OutputStream stream) {
```

Comment:
yeah, I thought I'll keep a class since we are planning to have other exporters in the future, but I guess .. when we'll have them we'll create a class 😃